RecoTracker/TkHitPairs/plugins/HitPairEDProducer.cc

```diff
@@ -290,8 +293,12 @@ void HitPairEDProducer::fillDescriptions(edm::ConfigurationDescriptions& descrip
 }
 
 void HitPairEDProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
-  edm::Handle<bool> hclusterCheck;
-  iEvent.getByToken(clusterCheckToken_, hclusterCheck);
+  bool clusterCheckOk = true;
+  if(!clusterCheckToken_.isUninitialized()) {
```

Comment:
shouldn't this instead rely on a ``` bool haveClusterCheck = clusterCheckTag.label() != "" ``` (computed at construction).
Otherwise it seems like we allow other undefined behavior (can a token be uninitialized for a non-empty label? .. or if the product was not added to the event in a given event)